plugin/src/main/java/org/opensearch/ml/task/MLPredictTaskRunner.java

```diff
@@ -148,6 +148,16 @@ public void dispatchTask(
                 if (FunctionName.isAutoDeployEnabled(autoDeploymentEnabled, functionName)) {
                     try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
                         mlModelManager.getModel(modelId, ActionListener.runBefore(ActionListener.wrap(model -> {
+                            if (!checkModelAutoDeployEnabled(model)) {
+                                log.info("Auto deployment disabled for model {}, please deploy model first", modelId);
+                                listener
+                                    .onFailure(
+                                        new IllegalArgumentException(
+                                            "Model not ready yet. Please run this first: POST /_plugins/_ml/models/" + modelId + "/_deploy"
```

Comment:
We are assuming customer is using dev tools.

May be set a generic message like: "The model is not deployed yet. Please deploy the model first." ?